lib/data_hygiene/bulk_organisation_updater.rb

```diff
@@ -0,0 +1,66 @@
+module DataHygiene
+  class BulkOrganisationUpdater
+    def initialize(filename)
+      @filename = filename
+    end
+
+    def call
+      CSV.foreach(filename, **CSV_OPTIONS) do |row|
+        process_row(row)
+      end
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+  private
+
+    attr_reader :filename
+
+    CSV_OPTIONS = {
+      headers: true,
+    }.freeze
+
+    def process_row(row)
+      document = find_document(row)
+      new_lead_organisations = find_new_lead_organisations(row)
+      new_supporting_organisations = find_new_supporting_organisations(row)
+
+      puts "#{document.slug}: #{new_lead_organisations.map(&:slug).join(', ')} (#{new_supporting_organisations.map(&:slug).join(', ')})"
+      update_document(document, new_lead_organisations, new_supporting_organisations)
+    end
+
+    def find_document(row)
+      Document.find_by!(slug: row.fetch("Slug"))
+    end
+
+    def find_organisations(row, column)
+      column_data = row.fetch(column)
+      return [] unless column_data
+
+      column_data
+        .split(",")
+        .map { |slug| Organisation.find_by!(slug: slug.strip) }
+    end
+
+    def find_new_lead_organisations(row)
+      find_organisations(row, "New lead organisations")
+    end
+
+    def find_new_supporting_organisations(row)
+      find_organisations(row, "New supporting organisations")
+    end
+
+    def update_document(document, new_lead_organisations, new_supporting_organisations)
+      edition = document.latest_edition
+
+      edition.update(
+        lead_organisations: new_lead_organisations,
+        supporting_organisations: new_supporting_organisations,
+      )
+
+      PublishingApiDocumentRepublishingWorker.perform_async(document.id)
```

Comment:
Checking if a change has been made would be good around this point, as if no change has been made, you don't need to republish the document.

I can imagine this coming in useful if you process a file, find an error part way through, then process the fixed file.

Reply:
That's a good idea! Will add that 👍

Reply:
I've amended the commit with that and added a test.

Reply:
:+1: